packages/utils/src/lib/reports/md/table.ts

```diff
@@ -7,46 +10,42 @@ const alignString = new Map<Alignment, string>([
   ['r', '--:'],
 ]);
 
+function tableRow(rows: (string | number)[]): string {
+  return `|${rows.join('|')}|`;
+}
+
 /**
  * | Table Header 1  | Table Header 2 |
  * | --------------- | -------------- |
  * |  String 1       |  1             |
  * |  String 1       |  2             |
  * |  String 1       |  3             |
  */
-export function tableMd(
-  data: (string | number)[][],
-  align?: Alignment[],
-): string {
-  if (data.length === 0) {
+export function tableMd<T extends Table>(data: T): string {
+  const { rows = [], alignment } = data;
+  if (rows.length === 0) {
     throw new Error("Data can't be empty");
   }
-  const alignmentSetting = align ?? data[0]?.map(() => 'c');
-  const tableContent = data.map(arr => `|${arr.join('|')}|`);
-  const alignmentRow = `|${alignmentSetting
-    ?.map(s => alignString.get(s))
-    .join('|')}|`;
-  return (
-    tableContent[0] +
-    NEW_LINE +
-    alignmentRow +
-    NEW_LINE +
-    tableContent.slice(1).join(NEW_LINE)
-  );
-}
 
-export function tableHtml(data: (string | number)[][]): string {
-  if (data.length === 0) {
-    throw new Error("Data can't be empty");
-  }
+  const stringArr = tableToFlatArray(data);
+
+  const allCenterAlignments = (
+    typeof rows.at(0) === 'string'
+      ? Array.from({ length: rows.length })
+      : Object.keys(rows.at(0) ?? {})
+  ).map(() => 'c' as Alignment);
```

Comment:
Using `as` loses type-safety, `'any random string' as Alignment` would compile.

```suggestion
  ).map((): Alignment => 'c');
```